### PR TITLE
chore: clean templates API to use pagination [DET-3754]

### PR DESCRIPTION
**Pull request by ybt195**
## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: proto/src/determined/api/v1/template.proto [hunk 1]
```diff
@@ -9,13 +9,31 @@ import "determined/template/v1/template.proto";
 
 // Get a list of templates.
 message GetTemplatesRequest {
+    // Sorts templates by the given field.
+    enum SortBy {
+        // Returns experiments in an unsorted list.
```
Comment (sidneyw) [inline proto/src/determined/api/v1/template.proto:14]: non-blocking: replace experiments with templates

Comment (hamidzr) [reply proto/src/determined/api/v1/template.proto]: I actually can't push to the source branch to fix this. merge this as is?

Comment (sidneyw) [reply proto/src/determined/api/v1/template.proto]: I'd rather you close this PR, pull the source branch, add the fixes, open a new PR and link it to this one for context.

**Comment by hamidzr** [comment]
closing in favor of https://github.com/determined-ai/determined/pull/1006

**Pull request closed without merging**